contents/testing/security-testing.md

```diff
@@ -0,0 +1,144 @@
+</pic>
+</center>
+
+<br>
+
+To understand this in more detail, we can deconstruct the application of security testing into its manifestation in different stages of software development. The following series of steps taken from the OWASP Testing Guide[^1] provide general applications of security testing in a generic SDLC.
+
+### Step 1: Before Development
+
+#### 1.1 Define an SDLC
+
+Before application development starts an adequate SDLC must be defined where security is inherent at each stage. 
+
+#### 1.2 Review Policies and Standards
+
+Ensure that there are appropriate policies, standards, and documentation in place. Documentation is extremely important as it gives development teams guidelines and policies that they can follow. An example of one such documentation is the OWASP Secure Coding Practices Quick Refernce Guide[^6].
+
+People can only do the right thing if they know what the right thing is.
+
+If the application is to be developed in Java, it is essential that there is a Java secure coding standard. If the application is to use cryptography, it is essential that there is a cryptography standard. No policies or standards can cover every situation that the development team will face. By documenting the common and predictable issues, there will be fewer decisions that need to be made during the development process.
+
+### Step 2: During Definition and Design
+
+#### 2.1 Review Security Requirements
+
+Security requirements define how an application works from a security perspective. It is essential that the security requirements are tested. Testing in this case means testing the assumptions that are made in the requirements and testing to see if there are gaps in the requirements definitions.
+
+For example, if there is a security requirement that states that users must be registered before they can get access to a particular section of a website, does this mean that the user must be registered with the system or should the user be authenticated? Ensure that requirements are as unambiguous as possible.
+
+#### 2.1 Review Design & Architecture
+
+Applications should have a documented design and architecture. This documentation can include models, textual documents, and other similar artifacts. It is essential to test these artifacts to ensure that the design and architecture enforce the appropriate level of security as defined in the requirements.
+
+Identifying security flaws in the design phase is not only one of the most cost-efficient places to identify flaws, but can be one of the most effective places to make changes. For example, if it is identified that the design calls for authorization decisions to be made in multiple places, it may be appropriate to consider a central authorization component. If the application is performing data validation at multiple places, it may be appropriate to develop a central validation framework (ie, fixing input validation in one place, rather than in hundreds of places, is far cheaper).
```

Comment:
Perhaps the example given can be more simple? I'm not sure if readers would understand what `central authoritsation component` means

Reply:
Maybe if I rephrase to "...consider having a central component to be responsible for all authorisation"?